### PR TITLE
Now handles CompletableFuture coming back from an un batched DF

### PR DESCRIPTION
**Pull request by bbakerman**
See #672 and #683

### CODE REVIEW

File: src/main/java/graphql/execution/batched/BatchedExecutionStrategy.java [hunk 1]
```diff
@@ -201,6 +202,52 @@ private void executeImpl(ExecutionContext executionContext,
 
     }
```
Comment (bbakerman) [inline src/main/java/graphql/execution/batched/BatchedExecutionStrategy.java]: Mostly moved this up in the file so its near where it gets called and tweaked the DF fetching to use the Async code

**Comment by bbakerman** [comment]
Ok so your original PR fixed this @andimarek but I think there is some refactor value in this one as well reagarding Async and using it commonly in a few places

File: src/main/java/graphql/execution/Async.java [hunk 2]
```diff
@@ -82,4 +84,39 @@
...
+
+    /**
+     * This combines a list of CompletableFuture each promising T into a CompletableFuture promising a list of T
+     *
+     * @param listOfFutures the list of futures
+     * @param <T> for two
+     *
+     * @return a CompletableFuture of List T
+     */
+    public static <T> CompletableFuture<List<T>> combineAllOf(List<CompletableFuture<T>> listOfFutures) {
```
Comment (andimarek) [inline src/main/java/graphql/execution/Async.java]: this is not needed: please use `each(List<CompletableFuture<U>> futures)`

File: src/main/java/graphql/execution/Async.java [hunk 3]
```diff
@@ -82,4 +84,39 @@
...
+
+    /**
+     * Turns an object T into a CompletableFuture if its not already
+     *
+     * @param t   - the object to check
+     * @param <T> for two
+     *
+     * @return a CompletableFuture
+     */
+    public static <T> CompletableFuture<T> toCompletableFuture(T t) {
```
Comment (andimarek) [inline src/main/java/graphql/execution/Async.java]: good idea :-)

**Pull request merged**